src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -124,8 +129,11 @@ run scheduler _origDir plugins captureFp = flip E.catches handlers $ do
             reactorFunc = react $ reactor rin diagIn
 
         let errorHandler :: Scheduler.ErrorHandler
-            errorHandler lid code e =
+            errorHandler (Just lid) code e =
               Core.sendErrorResponseS (Core.sendFunc lf) (J.responseId lid) code e
+            errorHandler Nothing _code e =
+              Core.sendErrorShowS (Core.sendFunc lf) e
```

Comment:
This is a hack to be able to send "diagnostics" from `runActionWithContext`. It shows errors that occur during project initialisation to the user via a textbox